session/session_test.go

```diff
@@ -5813,3 +5813,51 @@ func (s *testSessionSuite) TestSameNameObjectWithLocalTemporaryTable(c *C) {
 			"  `cs1` int(11) DEFAULT NULL\n" +
 			") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin"))
 }
+
+func (s *testSessionSuite) TestOptimizedDeleteYourWrites(c *C) {
```

Comment:
Seems this test also passes without optimization of delete your writes.

Reply:
This is used to verify the optimization would not break consistency. To verify if the dangling `delete` is optimized, we may need to check the mvcc version and such test cases are easier to implement in utf.